### PR TITLE
Bug fixes1

### PR DESCRIPTION
**Pull request by tru36rit**

### CODE REVIEW

File: src/components/Movie.js [hunk 1]
```diff
@@ -5,14 +5,36 @@ import MovieInfo from './MovieInfo';
...
   handleMouseClick = () =>
-    this.setState(prevState => ({ isHovering: !prevState.isHovering }));
+    this.setState(prevState => ({ isClicked: !prevState.isClicked }));
+
+  // The api returns objects which may or may not have the properties
+  // we want to pull off.  Therefore, we can't just try to pass down properties
+  // from this.props, we have to make a new object based upon this.props
+  // which handles the absence. Furthermore some properties are nested e.g.
+  // "Ratings", so we can't just assign an empty string as a value.
+  handlePropertyExistence(checkedObj, propertyList) {
```
Comment (ghaynes1292) [inline src/components/Movie.js]: move this elsewhere, and don't use reduce. If you want to modify ratings, modify ratings

File: src/utils/omdbApi.js [hunk 2]
```diff
@@ -6,18 +6,49 @@ function makeUrl(queryStr, parameter, key = OMDB_KEY) {
...
-  return axios.get(url);
+  try {
+    let url = makeUrl(queryStr, parameter);
+    return axios.get(url);
+  } catch (error) {
+    console.error('Error in apiFetch:', error);
+  }
+}
+
+function fixRatings(movie) {
```
Comment (ghaynes1292) [inline src/utils/omdbApi.js]: I would make this a function that returns what should be in the ratings field, then when calling it do something like:
```
{
  ...movie,
  ratings: newRatings(movie)
}
```

Comment (tru36rit) [reply src/utils/omdbApi.js]: Ok I see.  Seems simpler, thx.

Comment (tru36rit) [reply src/utils/omdbApi.js]: Done. (?)

**Pull request merged**